lib/puppet/type/acl/ace.rb

```diff
@@ -29,7 +29,11 @@ def initialize(permission_hash, provider = nil)
       self.id = permission_hash['id']
       @mask = permission_hash['mask']
       self.rights = permission_hash['rights']
-      self.type = permission_hash['type']
+      if permission_hash['type'] && !permission_hash['perm_type']
+        self.perm_type = permission_hash['type']
+      else
+        self.perm_type = permission_hash['perm_type']
+      end
```

Comment:
I always prefer to limit the number of tickets by giving the user feedback when they are doing things wrong so there is no question that they are doing something incorrect.